pkg/services/ngalert/state/manager_test.go

```diff
@@ -864,3 +867,107 @@ func TestProcessEvalResults(t *testing.T) {
 		})
 	}
 }
+
+func TestStaleResultsHandler(t *testing.T) {
+	evaluationTime, err := time.Parse("2006-01-02", "2021-03-25")
```

Comment:
This `evaluationTime` is already far in the past. It could be something closer the double value of the rule interval and even better having a mockable `timeNow()` function for more deterministic tests.

Reply:
The timestamp here shouldn't matter since the expected test cases are all done relative to evaluationTime. I like having the timestamp be known for easier debugging but if you've got an improvement I would be happy to review it. There's probably an opportunity for us to make a couple utilities/helpers along those lines to make our tests easier to write.

Reply:
I'm also fine also leaving it as is; I don't insist.

We don't have a library but for such cases we define:
[var timeNow = time.Now](https://github.com/grafana/grafana/blob/7815ed511f6bb0fb86e7274f91d36987ccf648ce/pkg/services/sqlstore/alert.go#L15)
and use this instead so that in test we can mock it to return something deterministic:
https://github.com/grafana/grafana/blob/7815ed511f6bb0fb86e7274f91d36987ccf648ce/pkg/services/sqlstore/alert_test.go#L14-L26